### ISSUE TITLE
[feature request] escape tree entry into its own tree

### ISSUE THREAD

**Issue by KamelittaOida** [piroor/treestyletab#2964]
I've gone through all the options and all shortcut keys (41 keys), and I don't think that feature is there.

Is it possible to make so that sub-trees can be moved into their own trees?

![treestyle1](https://user-images.githubusercontent.com/29007971/125068130-6c8fd500-e0b5-11eb-8825-6671bcee9422.png)

**Comment by piroor** [comment]
Sorry I couldn't understand what is the expected feature. Could you describe more with details?
I thought that you meant: promote the 7th descendant tab to the root parent and make all preceding tabs descendants of the 7th tab. Is this understanding correct?

**Comment by KamelittaOida** [comment]
> promote the 7th descendant tab to the root parent and make all preceding tabs descendants of the 7th tab. Is this understanding correct?

promote the 7th descendant tab **_to be a_** root parent and make all **_succeeding_** tabs, within its tree up to the next tree, own descendants of the 7th tab. 

ok wait. I will make a screen shot

**Comment by KamelittaOida** [comment]
well I guess what I'm basically asking for is to change position hierarchically but not as a tab. No idea how technically possible this is. (move tab on TST laterally but not vertically).
Or: convert any tab into a potential parent for any tabs below it, up until the next tree.
I guess I want to save on selecting tabs and having to move them with the mouse etc. Just a keyboard way of making parents/trees.

sry for the music. mute please. 
https://vimeo.com/573015007
pw:
aaaaaaaaaaaa

**Comment by irvinm** [comment]
This is definitely possible but might be better for a helper addon.  It is just reviewing the relationships and changing the parent-child relationships based on the action desired.

I think there would need to be support for:

- Promote tab (move tab 1 level higher) - if children, reassign them to parent of tab being moved?
- Demote tab (move tab 1 level lower - if possible)
- Promote tree (move tree 1 level higher)
- Demote tree (move tree 1 level lower - if possible)
- Promote tab to root (move tab to root level) - if children, reassign them to parent of tab being moved?
- Demote tab to bottom (move tab to deepest nesting)
- Promote tree to root (move tree to root level)
- Demote tree to bottom (move tree to deepest nesting)

There would also probably need to be an option if the direction should be "up" or "down" of current location in case of the "root" use cases. (From the current tree, move the desired tab\tree above the current tree or below the current tree)  I would also think that all of the "Demote" options wouldn't be very useful ... so maybe just do the "Promote" actions.

**Comment by KamelittaOida** [comment]
>\1. Promote tab (move tab 1 level higher) - if children, reassign them to parent of tab being moved?
>\5. Promote tab to root (move tab to root level) - if children, reassign them to parent of tab being moved?

yes, I think these are the two most important ones. Because sometimes I want to promote a tab, but it's just too much with having to select them (then some tabs get loaded etc) and fumble around with dropping them into the tree I want.
It's very seldom I think about demoting a tab.

I think I've explained myself enough, but basically: a possibility to make any tab into a root parent with keyboard, but without the tab changing position in Firefox, just the hierarchy position in TST.

I think it will be easier to manage tabs that way: ctrl+shift+ page(up/down) to move tab into position. Then [shortcut to promote] tab into root parent or parent/sibling
